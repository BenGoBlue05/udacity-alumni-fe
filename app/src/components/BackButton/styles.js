import styled from 'styled-components';
import Button from 'grommet-udacity/components/Button';
import Box from 'grommet-udacity/components/Box';
import LinkPrevious from 'grommet-udacity/components/icons/base/LinkPrevious';

export const FabButton = styled(Button)`
  stroke: white !important;
`;

export const FabIcon = styled(LinkPrevious)`
  font-size: 1.6rem;
  stroke: white !important;
`;

export const FabContainer = styled(Box)`
  z-index: 1000;
  position: absolute;
  top: 0;
  left: 0;
  margin: 1.5em;
  @media screen and (max-width: 768px) {
    margin: 1em;
  }
`;

export const Fab = styled(Box)`
  background-color: #89A2B5;
  display: flex;
  align-items: center;
  justify-content: center;
  border-radius: 50%;
  height: 60px;
  width: 60px;
  box-shadow: 0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24);
  transition: all 0.3s cubic-bezier(.25,.8,.25,1);
  &:hover {
    box-shadow: 0 14px 28px rgba(0,0,0,0.25), 0 10px 10px rgba(0,0,0,0.22);
  }
  title {
    color: black !important;
  }
  @media screen and (max-width: 768px) {
    height: 40px;
    width: 40px;
  }
`;
