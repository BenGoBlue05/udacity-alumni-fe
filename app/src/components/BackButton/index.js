import React, { PropTypes, Component } from 'react';
import { FabContainer, Fab, FabButton, FabIcon } from './styles';

class BackButton extends Component {
  render() {
    return (
      <FabContainer>
        <Fab>
          <FabButton
            icon={
              <FabIcon
                a11yTitle="Back button"
                a11yTitleId="back-button-floating-button-icon"
              />
            }
            onClick={() => this.context.router.goBack()}
            a11yTitle="Add Review Floating"
            a11yTitleId="add-review-floating-button"
          />
        </Fab>
      </FabContainer>
    );
  }
}

BackButton.contextTypes = {
  router: PropTypes.object.isRequired,
};

export default BackButton;
