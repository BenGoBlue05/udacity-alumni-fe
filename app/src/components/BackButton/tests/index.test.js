import BackButton from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

describe('<BackButton />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <BackButton />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
