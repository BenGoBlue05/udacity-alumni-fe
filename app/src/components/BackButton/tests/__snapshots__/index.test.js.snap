exports[`<BackButton /> should render with default props 1`] = `
<Styled(Box)>
  <Styled(Box)>
    <Styled(Button)
      a11yTitle="Add Review Floating"
      a11yTitleId="add-review-floating-button"
      icon={
        <Styled(LinkPrevious)
          a11yTitle="Back button"
          a11yTitleId="back-button-floating-button-icon" />
      }
      onClick={[Function onClick]} />
  </Styled(Box)>
</Styled(Box)>
`;
